user/authentication.py

```diff
@@ -0,0 +1,35 @@
+import jwt
+from condominium_manager_server.settings import SECRET_JWT
+from rest_framework import authentication, status
+from rest_framework.exceptions import AuthenticationFailed
+
+from user.models import User
+
+
+class JWTCustomAuthentication(authentication.BaseAuthentication):
+    def authenticate(self, request):
+        auth = request.META.get('HTTP_AUTHORIZATION', '')
+        if not auth:
+            return None
+
+        try:
+            access_token = auth.split(' ')[1]
```

Comment:
**Suggestion:** In the JWTCustomAuthentication class, the 'authenticate' method could be improved by handling the case where the 'Authorization' header is not in the expected format (i.e., "Bearer <token>"). Currently, if the header doesn't contain a space, the 'split' method will only return a single element list, causing an IndexError when trying to access the second element.
```suggestion
        auth = request.META.get('HTTP_AUTHORIZATION', '')
        if not auth:
            return None
        try:
            auth_type, access_token = auth.split(' ')
            if auth_type.lower() != 'bearer':
                raise AuthenticationFailed('Invalid token prefix')
        except ValueError:
            raise AuthenticationFailed('Invalid authorization header format')
```